src/field/field3d.cxx

```diff
@@ -263,13 +233,16 @@ Field3D & Field3D::operator=(const Field3D &rhs) {
 
   TRACE("Field3D: Assignment from Field3D");
   
+  copyFieldMembers(rhs);
+
   // Copy the data and data sizes
-  fieldmesh = rhs.fieldmesh;
-  nx = rhs.nx; ny = rhs.ny; nz = rhs.nz; 
-  
-  data = rhs.data;
+  nx = rhs.nx;
+  ny = rhs.ny;
+  nz = rhs.nz;
 
-  setLocation(rhs.location);
+  ASSERT1(fieldsCompatible(*this, rhs));
```

Comment:
I think this be done immediately after `TRACE` and before `copyFieldMembers`. It seems likely this test will always pass in this location

Reply:
If `copyFieldMembers` is working correctly then this test should indeed always pass. So it's only a sanity check that `copyFieldMembers` and `areFieldsCompatible` haven't somehow become inconsistent. It's probably better to check that in a unit test than here...